File: src/test/java/com/doestheohavethisbook/controllers/HelloControllerTests.java
package com.doestheohavethisbook.controllers;

import org.junit.Test;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

import static org.junit.Assert.assertThat;
import static org.hamcrest.core.Is.is;
import static org.hamcrest.CoreMatchers.*;

public class HelloControllerTests {
    @Test
    public void whenHelloCalled_thenReturnsHelloWorld() {
        HelloController controller = new HelloController();
        ResponseEntity<String> result = controller.hello();

        assertThat(result.getStatusCode(), is(equalTo(HttpStatus.OK)));
        assertThat(result.getBody(), is("Hello, World!"));
    }
}

File: src/test/java/com/doestheohavethisbook/controllers/HelloControllerTests.kt
package com.doestheohavethisbook.controllers

import junit.framework.TestCase.assertEquals
import org.junit.Test
import org.springframework.http.HttpStatus

class HelloControllerTests {
    @Test
    fun whenHelloCalled_thenReturnsHelloWorld() {
        val controller = HelloController()
        val result = controller.hello()

        assertEquals(HttpStatus.OK, result.getStatusCode())
        assertEquals("Hello, World!", result.getBody())
    }
}
